### ISSUE TITLE
Prune import statement symbols

### ISSUE THREAD

**Issue by Sir4ur0n** [haskell/haskell-language-server#215]
It's been a long time since my last issue :stuck_out_tongue: 

Currently HLS considers each `import` statement as a symbol, which has several minor inconveniences, e.g.:
* `Go to symbol` is spammed with import statements, which - I think - almost nobody will ever use
* `Go to Next Member` stops at each `import` statement, while the whole point is to quickly navigate in the file

I just checked how it's done e.g. with Java files: import statements are not even symbols.

Concrete example:
```haskell
module Foo where

import Prelude
import Data.Text
import GHC.Generics
import GHC.Arr

foo :: ()
foo = ()

bar :: ()
bar = ()

data Foo

newtype Bar = Bar Foo

class Baz a

instance Baz Foo
```

![image](https://user-images.githubusercontent.com/1204125/88063423-4e5e5b80-cb6a-11ea-8527-9011d425d0ef.png)

As you can see, there is a symbol for the module declaration, another for "the group of imports" (which I find awesome), in my opinion there is no benefit to also consider each import statement as a symbol.

What do you think?

**Comment by xsebek** [comment]
I will try to find the relevant code (@fendor said it's in `ghcide`) and make a PR. 🙂

**Comment by xsebek** [comment]
Here is the relevant PR on ghcide that introduced the "imports" group: https://github.com/haskell/ghcide/pull/443

One way to solve it would be to simplify the code and only keep the "imports" group discarding the individual imports.

I am trying to find in the docs if we could customize this in HLS or `vscode-haskell` plugin just for this selection.

**Comment by xsebek** [comment]
**TLDR:** type `:` after `@`. 😉 

---

We discussed this with @fendor and we agreed **not** to implement this:

* the side panel works well and shows the hierarchy of imports grouped in "imports"
* there does not seem to be a way on HLS side to tell whether this is coming from the side panel or the "go to symbol" selection
* this does not seem to be highly requested (no one implemented this in 6 years) and its possible someone[<sup>*</sup>](https://xkcd.com/1172/) even depends on this behavior

We checked what Python LSP does and it is worse:

* The outline in the side panel is not grouped and `from X import Y` shows up as if you defined `Y` yourself (see `Dict`):
  <img width="250" alt="Image" src="https://github.com/user-attachments/assets/d521310b-3b00-4145-83a3-d0b0d32cc66b" />
* The "go to symbol" similarly lists all imports
  <img width="250" alt="Image" src="https://github.com/user-attachments/assets/4d34c826-54cb-486c-9e56-c3219cdd0c30" />

But finally, searching through the Microsoft documentation revealed this amazing knowledge:
> _You can navigate symbols inside a file with ⇧⌘O. By typing `:` the symbols will be grouped by category. Press Up or Down and navigate to the place you want._

And sure enough that shows functions first before imports:

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/22a8df99-9bfa-4a8e-8075-14b2cc34fbb8" />
